### PR TITLE
Remove the `iui-anchor` mixin.

### PR DESCRIPTION
**Pull request by Ben-Pusey-Bentley**
## Changes

Removes the following `iui-anchor` mixin from the css package.

The `iui-anchor` mixin has instead had it's styles all added to the `iui-anchor` class in the `itwinui-css` package. This is breaking change for all of the components that previously used this mixin in their styles, as they now need to have the `iui-anchor` class added onto them to be styled correctly. The `css-workshop` has been updated to reflect this change. Similarly, the `itwinui-react` package has been updated to use this new convention. There is still a breaking change to the custom elements for the `Footer` component, however, since those links need to have a class defined inside of them to have the correct styling. The `react-workshop` has also been updated to reflect this.

## After-PR TODO(s)
- [ ] Add support for `<button>` and `<a>` elements to both the `AlertAction` and `ToastPresentation` components, as both components should be able to have either functionality. (see [thread](https://github.com/iTwin/iTwinUI/pull/1915#discussion_r1528845986))

## Testing

Ran tests with both the `css-workshop` and `react-workshop` to make sure that no unintended visual changes had been made.

## Docs

Added changeset for both `itwinui-css` and `itwinui-react`.

### CODE REVIEW

**Review by mayank99 — COMMENTED** [review]
Can you split this up into two separate PRs for anchor and globals?

**Comment by Ben-Pusey-Bentley** [comment]
> Can you split this up into two separate PRs for anchor and globals?

Okay, I'll make this one the PR for anchor then.

File: apps/react-workshop/src/Footer.stories.tsx
Comment (mayank99) [inline apps/react-workshop/src/Footer.stories.tsx:114]: This is not acceptable. The class names are an implementation detail that should not be exposed to the user. (The "_" prefix indicates that it's considered private)

File: packages/itwinui-css/src/anchor/mixins.scss
Comment (mayank99) [inline packages/itwinui-css/src/anchor/mixins.scss:5]: These other mixins should also be removed. Everything should be controllable using plain CSS/HTML.

File: packages/itwinui-react/src/core/Footer/Footer.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Footer/Footer.tsx:154]: You don't need to go through such great lengths to add the class here. Using the `Anchor` component will suffice.

File: packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx:236]: `Anchor` component could be used here as well

File: .changeset/green-cheetahs-rush.md
Comment (mayank99) [inline .changeset/green-cheetahs-rush.md:1]: Ideally this change should be transparent to react users.

File: apps/react-workshop/src/Footer.stories.tsx
Comment (Ben-Pusey-Bentley) [reply apps/react-workshop/src/Footer.stories.tsx:114]: Okay, then I'll replace the `<a>` here with `Anchor`.

File: packages/itwinui-react/src/core/Footer/Footer.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Footer/Footer.tsx:154]: Will do.

File: packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx:236]: Will do.

File: .changeset/green-cheetahs-rush.md
Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Ideally, but anyone who is using an `<a>` in a `Footer` custom element will have to replace that with an `Anchor` with how the changes are currently implemented. Could add the styles from the `iui-anchor` class to the `> a` selector for the `iui-legal-footer-item` class, but reusing styles like that feels like it slightly defeats the purpose of removing the mixin.

File: .changeset/green-cheetahs-rush.md
Comment (rohan-kadkol) [inline .changeset/green-cheetahs-rush.md:3]: After #1910, `itwinui-css` would need a separate changeset.

File: packages/itwinui-css/src/alert/alert.scss
Comment (rohan-kadkol) [inline packages/itwinui-css/src/alert/alert.scss:64]: Is `.iui-anchor` needed in combination with `.iui-alert-link`? Or is `.iui-alert-link` sufficient?

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (rohan-kadkol) [inline packages/itwinui-react/src/core/Alert/Alert.tsx:139]: Should `.iui-anchor` only be applied when the `as` is an `<a>`?

File: packages/itwinui-css/src/alert/alert.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/alert/alert.scss:64]: It is needed in combination here because using just `.iui-alert-link` causes the style to be overwritten by the style in `.iui-anchor`, so a higher specificity is required.

File: .changeset/green-cheetahs-rush.md
Comment (mayank99) [reply .changeset/green-cheetahs-rush.md:1]: Well we can't suddenly ask our users to use `<Anchor>` instead of `<a>`. That's a breaking change.

What we might be able to do though, is clone the node and modify it? It's not ideal, but backwards compatibility is important.

Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Okay, I'll take a look at cloning the node then.

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Alert/Alert.tsx:139]: The type on that `forwardRef` indicates that the `AlertAction` will always be an `<a>`, which is a little confusing with the `as` potentially being a `<button>`. I will investigate this and see if there are any issues with only applying the `.iui-anchor` class when `as` is `<a>`.

File: .changeset/green-cheetahs-rush.md
Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Added some code to the `<Footer>` that replaces the `<a>` added to custom content with `<Anchor>` in the latest [commit](https://github.com/iTwin/iTwinUI/pull/1915/commits/45c6861c0477f194b1f2e3955493ea8bdf78321b). Seems like it might be fragile in its current state.

File: apps/react-workshop/src/Footer.stories.tsx
Comment (Ben-Pusey-Bentley) [reply apps/react-workshop/src/Footer.stories.tsx]: Removed that `className` in the latest [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/45c6861c0477f194b1f2e3955493ea8bdf78321b)

File: .changeset/green-cheetahs-rush.md
Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:2]: Removed the `itwinui-react` from the changeset in this [commit](https://github.com/iTwin/iTwinUI/pull/1915/commits/45c6861c0477f194b1f2e3955493ea8bdf78321b), since the code has been updated to remove breaking changes for react users. [Related Comment](https://github.com/iTwin/iTwinUI/pull/1915#discussion_r1523450673)

File: .changeset/green-cheetahs-rush.md
Comment (mayank99) [reply .changeset/green-cheetahs-rush.md:1]: Yeah it looks pretty bad. I'd say lets just add some CSS for `.iui-legal-footer-item > a`. It should only be a few lines (we don't need everything from `iui-anchor`).

Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Alright, will do.

File: packages/itwinui-css/src/alert/alert.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/alert/alert.scss:64]: Done in latest [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/a6144abc9c3d1d69d0640a285709d1d72f47bd5b)

File: packages/itwinui-css/src/anchor/mixins.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/anchor/mixins.scss]: Done in latest [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/63c95e30856b494b4444148e0b90542656c6ddb9)

File: .changeset/green-cheetahs-rush.md
Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Done in this [commit](https://github.com/iTwin/iTwinUI/pull/1915/commits/45c6861c0477f194b1f2e3955493ea8bdf78321b) and high contrast fixed in [latest.](https://github.com/iTwin/iTwinUI/pull/1915/commits/63c95e30856b494b4444148e0b90542656c6ddb9)

File: packages/itwinui-css/src/alert/alert.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/alert/alert.scss:70]: Sorry, ignore the last comment here. Accidently put that comment on the wrong conversation.

Comment (rohan-kadkol) [reply packages/itwinui-css/src/alert/alert.scss:70]: To confirm, is that commit link correct? Because that commit has no changes for `alert.scss`.

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/alert/alert.scss:70]: Again, please ignore that comment. It was meant for this [conversation.](https://github.com/iTwin/iTwinUI/pull/1915#discussion_r1523450673)

Comment (rohan-kadkol) [reply packages/itwinui-css/src/alert/alert.scss:70]: Oops, I missed reading https://github.com/iTwin/iTwinUI/pull/1915#discussion_r1526290640 😅

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/alert/alert.scss:70]: All good. It was my mistake to begin with.

File: packages/itwinui-css/src/anchor/mixins.scss
Comment (mayank99) [reply packages/itwinui-css/src/anchor/mixins.scss]: For `iui-alert-link` and `iui-toast-anchor`, you can use the `underline` prop in React and the `data-iui-underline` attribute in HTML, instead of repeating the underline CSS.

Also you can add new data attribute selector to `.iui-anchor` to reduce the repetition around status colors. The HTML would then become something like `<a class="iui-anchor iui-alert-link" data-iui-status="positive">`.

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/anchor/mixins.scss]: Okay, will do.

File: packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx]: Done in latest [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/ecdd7cf46b7f29bc1f86b77f40b7f23364723ce5)

Comment (mayank99) [reply packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx]: You don't need to branch on the return value. I'm just suggesting this one line change:

```diff
- const defaultAs = !!props.href ? 'a' : !!props.onClick ? 'button' : 'span';
+ const defaultAs = !!props.href ? Anchor : !!props.onClick ? 'button' : 'span';
```

File: packages/itwinui-react/src/core/Footer/Footer.test.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Footer/Footer.test.tsx:40]: This is testing implementation details and is therefore brittle. You can instead do a partial match using `toHaveTextContent`:

```js
expect(element).toHaveTextContent(data[dataIndex].title);
```

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Footer/Footer.test.tsx:40]: Alright, will do.

File: packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx]: I tried that as a string and it didn't work for me. Guess I didn't realize that it had to be a type if it is a React component. Will fix.

**Review by AdamMeza-Bentley — APPROVED** [review]

File: packages/itwinui-css/src/anchor/mixins.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/anchor/mixins.scss]: Done in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/882a3b0dc56a27a8668d6b79dff78bdedeb4b9fd) Changed the `<ButtonBase>` to `<Anchor>` in the `Alert` component to use the `underline` prop, but couldn't do so with `Toast`, since the `link` object that gets spread across the button there takes in `HTMLButtonProps`. Instead, passed in `data-iui-underline` to the button. Could also add an `underline` button to prop, but I decided against it since we are trying to limit changes to react users.

File: packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx]: Done in latest [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/4f086a8b25654918d9a640e46f5cc1ee1ba9735f)

File: packages/itwinui-css/src/alert/alert.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/alert/alert.scss]: The style ended up being completely removed in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/882a3b0dc56a27a8668d6b79dff78bdedeb4b9fd#diff-f4d2f744678b91bc8388e5cfed9d9326fa6f7061ae50ae71d5b542a8b09c3389) It was removed because of this [conversation.](https://github.com/iTwin/iTwinUI/pull/1915#discussion_r1526311053)

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Alert/Alert.tsx]: I can confirm that the visual tests expect the styles that we provide with the `.iui-anchor` class, even when the element is a `<button>`. It therefore makes sense to apply the class to the `AlertAction`, regardless of the `as` parameter.

File: packages/itwinui-css/src/anchor/anchor.scss
Comment (mayank99) [inline packages/itwinui-css/src/anchor/anchor.scss:37]: Nit: could you reorder this file so that these loose declarations (`color`, `cursor`, etc) are at the top of the class, and the `:hover` declarations are grouped, and the status colors are at the bottom?

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/anchor/anchor.scss:37]: Sure, will do.

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Alert/Alert.tsx:140]: nit:

```suggestion
      className={cx('iui-button-base', 'iui-alert-link', className)}
```

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Alert/Alert.tsx:138]: By making this change we lose out on the useful behaviors provided by `ButtonBase`.

I'm thinking we should use `Buttonbase` inside `Anchor` too. Or maybe we need a generic component called `Action` (or `ButtonOrLink`) 🤔

(Not sure of the best solution yet, and doesn't need to be resolved in this PR)

File: packages/itwinui-react/src/core/Toast/Toast.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Toast/Toast.tsx:289]: This should use `cx` for merging class names.

Also it should take into account `link.className`?

```jsx
<ButtonBase
  {...link}
  className={cx('iui-anchor', 'iui-toast-anchor', link.className)}
```

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Alert/Alert.tsx:138]: I can test out using `ButtonBase` inside of the `Anchor`, and see if there are any issues with that.

File: packages/itwinui-react/src/core/Toast/Toast.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Toast/Toast.tsx:289]: Alright, will do.

File: packages/itwinui-react/src/core/Footer/Footer.test.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Footer/Footer.test.tsx]: Done in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/34d0dad5d00a597dd17d66d6706c4f98dd71c73f)

File: packages/itwinui-css/src/anchor/anchor.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/anchor/anchor.scss:12]: Done in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/aae66014d51e43dc6228a8bb087f946daf705f44)

File: packages/itwinui-react/src/core/Toast/Toast.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Toast/Toast.tsx]: Done in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/bf9f560bd5cdba19aa3422624f98b7086412affa)

File: .changeset/green-cheetahs-rush.md
Comment (mayank99) [inline .changeset/green-cheetahs-rush.md:1]: Changeset can be reworded and split into two:
- One changeset (major) should clear mention the breaking HTML change (needing to pass `iui-anchor` explicitly)
- The other changeset (minor) can talk about the new status attribute support in `iui-anchor`

Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Will do.

Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Done in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/b2d784d01ab0a3cb57ccc6b0c9e89dc95c2e3f52) They may need some more rewording, as they both feel a little long in their current state.

Comment (mayank99) [reply .changeset/green-cheetahs-rush.md:1]: If you feel they need rewording, shouldn't you try to improve them before pushing? 

For a good example you can look at this: https://github.com/iTwin/iTwinUI/blob/rohan/table-header-scrollable-on-drag/.changeset/curvy-pugs-agree.md

Long isn't necessarily a problem, but it needs to be clear and easy to parse, without being overly wordy.

Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Made updates to both changesets in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/584dbdd0c7392603b8496ae89486eda4bf3add1f)

Comment (mayank99) [reply .changeset/green-cheetahs-rush.md:1]: It still feels overly wordy and unclear.

Here, I rewrote them to be make them more clear:

major:
> `iui-anchor` must now be explicitly added to `<a>` elements, in addition to the previous `iui-alert-link`, `iui-breadcrumbs-content`, `iui-tag-basic`, and `iui-toast-anchor` classes.
> 
> ```diff
> - <a class="iui-alert-link">…</a>
> + <a class="iui-alert-link iui-anchor">…</a>
> ```
> ```diff
> - <a class="iui-breadcrumbs-content">…</a>
> + <a class="iui-breadcrumbs-content iui-anchor">…</a>
> ```
> ```diff
> - <a class="iui-tag-basic">…</a>
> + <a class="iui-tag-basic iui-anchor">…</a>
> ```
> ```diff
> - <a class="iui-toast-anchor">…</a>
> + <a class="iui-toast-anchor iui-anchor">…</a>
> ```
> 
> Additionally, all `<a>` elements within `iui-legal-footer` elements will need the `iui-anchor` class.
> 
> ```diff
>   <li class="iui-legal-footer-item">
> -  <a>…</a>
> +  <a class="iui-anchor">…</a>
>   </li>
> ```

minor:

> `iui-anchor` can now be styled with status colors using a new `data-iui-status` attribute.
> 
> ```html
> <a class="iui-anchor" data-iui-status="negative">…</a>
> ```
> 
> Possible values include: `"positive"`, `"negative"` and `"informational".

Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Okay, thank you. I will update the changesets.

Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:1]: Done in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/91e9232447652bc3aff673e4ba80d8977bf45bc6) Also added the `warning` value to the possible values for `data-iui-status`.

**Review by mayank99 — APPROVED** [review]
LGTM. I would also wait for approval from @rohan-kadkol

File: .changeset/green-cheetahs-rush.md
Comment (rohan-kadkol) [inline .changeset/green-cheetahs-rush.md:5]: To confirm, are we intending to use the status only when the anchor is used inside another component? Or are we intending it to be used on standalone anchors too?

If we don't intend this to be used on standalone anchors, maybe we should mention this in the changeset?

Also, in general, should we mention something regarding the intended use-cases of this to prevent overruse of anchors with statuses?

File: packages/itwinui-css/src/footer/footer.scss
Comment (rohan-kadkol) [inline packages/itwinui-css/src/footer/footer.scss:39]: Is this the same as `&`? Maybe you did this for some specificity conflict?

File: apps/css-workshop/pages/tag.html
Comment (rohan-kadkol) [inline apps/css-workshop/pages/tag.html:736]: nit: Looks like there's some extra leading and trailing space.

image

image

I'm wondering how these weren't pointed out by the visual tests?

File: packages/itwinui-css/src/tag/tag.scss
Comment (rohan-kadkol) [inline packages/itwinui-css/src/tag/tag.scss:23]: Is this needed? Or just letting it (`@include mixins.iui-focus(…)`) come from the `.iui-anchor` is sufficient?

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (rohan-kadkol) [reply packages/itwinui-react/src/core/Alert/Alert.tsx:138]: Not sure why did you change the `ButtonBase` to `Anchor` here. Would just adding `.iui-anchor` along with the necessary data attributes like how you did for `Toast` not be sufficient?

image

File: apps/css-workshop/pages/tag.html
Comment (rohan-kadkol) [reply apps/css-workshop/pages/tag.html:736]: Actually, this can be be ignored since it's likely an issue on my side. Because [this PR's deployed version's `tag.html`](https://itwin.github.io/iTwinUI/1915/css/tag.html) shows up correctly.

File: packages/itwinui-css/src/tag/tag.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/tag/tag.scss:23]: It may not be. Let me check.

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (mayank99) [reply packages/itwinui-react/src/core/Alert/Alert.tsx:138]: `Anchor` is the correct way to do this. `Toast` should do the same

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Alert/Alert.tsx:138]: The primary reason for making the change was to give access to the `underline` prop, which currently only exists on `Anchor`. The reason that `Toast` doesn't do the same is because the `Toast` anchor takes in a `link` prop, which has `HTMLButtonProps` as a part of it's type. Changing that would introduce a breaking change for the react package.

File: packages/itwinui-css/src/tag/tag.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/tag/tag.scss]: Doesn't seem like it was. Removed in this [commit.](https://github.com/iTwin/iTwinUI/pull/1915/commits/73f14a143597afbe40d9972bdab2586317536b48)

File: packages/itwinui-css/src/footer/footer.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/footer/footer.scss:39]: `&` and `&:not(:hover)` are not the same. `&` is a rule that is always applied, while `&:not(:hover)` only applies it's styles when the element is not being hovered over. That change would create a conflict that does resolve correctly in this case, since the `:hover` style from `iui-anchor` has a higher specificity. However, I do not think it is wise to rely on the `:hover` rule winning the specificity battle for the footer links to be styled correctly.

File: .changeset/green-cheetahs-rush.md
Comment (Ben-Pusey-Bentley) [reply .changeset/green-cheetahs-rush.md:5]: The primary use case for this attribute is inside of the `Alert` and `Toast` components, but I don't think that means that developers should be cautioned from using this attribute with standalone anchors, if they feel like it. The attribute only changes the anchor's color, so I don't think overuse is much of a concern either. Mentioning the primary use case might make it more clear why we are adding this attribute, but I don't want to accidently make the changeset unclear as a result.

File: packages/itwinui-css/src/footer/footer.scss
Comment (rohan-kadkol) [reply packages/itwinui-css/src/footer/footer.scss:39]: > `&` and `&:not(:hover)` are not the same.

True, but I meant more like selecting something when it's not hovered (`&:not(:hover)`) would likely be the same as selecting that something directly without any pseudo classes (`&`), unless we *really* want to select it only when it's not hovered.

Upon rethinking it, are all those `text-decoration` styles required here ([code](https://github.com/iTwin/iTwinUI/pull/1915/files#diff-1cd601586d11db817d96d51d666331f1ea862f34ce736612b6a16204ed128583R39-R54))? Because I think it they all also from the anchor?

File: .changeset/green-cheetahs-rush.md
Comment (mayank99) [reply .changeset/green-cheetahs-rush.md:5]: This changeset is for the CSS package, which is really only used by us. We are not adding this as a prop in React, so there is no chance of misuse.

You can compare the download numbers of the recent versions:
- css: https://www.npmjs.com/package/@itwin/itwinui-react?activeTab=versions
- react: https://www.npmjs.com/package/@itwin/itwinui-react?activeTab=versions

File: packages/itwinui-css/src/footer/footer.scss
Comment (mayank99) [reply packages/itwinui-css/src/footer/footer.scss:39]: > > `&` and `&:not(:hover)` are not the same.
> 
> True, but I meant more like selecting something when it's not hovered (`&:not(:hover)`) would likely be the same as selecting that something directly without any pseudo classes (`&`), unless we _really_ want to select it only when it's not hovered.


`&:not(:hover)` was suggested by me. It is overriding the default user-agent styles of `<a>` element (which comes with an underline)

> Upon rethinking it, are all those `text-decoration` styles required here ([code](https://github.com/iTwin/iTwinUI/pull/1915/files#diff-1cd601586d11db817d96d51d666331f1ea862f34ce736612b6a16204ed128583R39-R54))? Because I think it they all also from the anchor?

`Anchor` may not always be used, like in the Custom Content Footer story

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (mayank99) [reply packages/itwinui-react/src/core/Alert/Alert.tsx:138]: Yeah lets handle this in a separate PR then. I think both Alert and Toast should correctly support both `<button>` and `<a>` because there are valid uses for both.

You can add an after-PR TODO in the PR description

File: packages/itwinui-css/src/footer/footer.scss
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-css/src/footer/footer.scss:39]: These styles are specifically being included because the `Footer` component allows for custom content. Removing these styles would break the styling for any custom content that does not use the `iui-anchor` class, which would introduce a breaking change to the react components for any users using custom content. As a result, it was decided to keep the styles here. Relevant conversation: [here](https://github.com/iTwin/iTwinUI/pull/1915#discussion_r1525481818)

File: packages/itwinui-react/src/core/Alert/Alert.tsx
Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/src/core/Alert/Alert.tsx:138]: Added a TODO to the PR description.

File: packages/itwinui-css/src/footer/footer.scss
Comment (rohan-kadkol) [reply packages/itwinui-css/src/footer/footer.scss:39]: Ohh, I see, I now understand why we need those styles.

In high contrast, the underline even shows up on hover. Is this expected? ([demo](https://itwin.github.io/iTwinUI/1915/react/?arg-high-contrast=true&story=footer--custom-content&theme=dark))

https://github.com/iTwin/iTwinUI/assets/45748283/adeda38d-5ae2-438f-9127-42ccd174b39d

Comment (mayank99) [reply packages/itwinui-css/src/footer/footer.scss:39]: I think that's ok. In a future PR, I would like to make all anchors have underlines on hover, even in high-contrast theme.

**Review by rohan-kadkol — APPROVED** [review]

**Pull request merged**
